### ISSUE TITLE
Ubuntu PPA misses latest version

### ISSUE THREAD

**Issue by NoTitleGamesOfficial** [lutris/lutris#4296]
When Installing on Ubuntu 20.04 using:
```
sudo add-apt-repository ppa:lutris-team/lutris
sudo apt update
sudo apt install lutris
```
I get version 0.5.10 which has this bug: https://github.com/lutris/lutris/issues/4256
In that thread they recommend to update to 0.5.10.1, but that one is not the latest here: https://launchpad.net/~lutris-team/+archive/ubuntu/lutris

**Comment by marhmm** [comment]
I can confirm it, PPA version is still 0.5.10. No update after 1.5 months.
I tried to contact PPA maintainer a few weeks ago but no reaction. Maybe you should try to contact them too? You can find a contact form on the site you posted.

**Comment by multimeric** [comment]
This is super annoying, but for what it's worth you can just download the DEB from the release page: https://github.com/lutris/lutris/releases/tag/v0.5.10.1. You might have to uninstall the existing version, and also `killall lutris` in case it's running in the background, but the whole process only takes a few seconds and it's fixed all the 0.5.9 bugs I've been hitting.

**Comment by NoTitleGamesOfficial** [comment]
> This is super annoying, but for what it's worth you can just download the DEB from the release page: https://github.com/lutris/lutris/releases/tag/v0.5.10.1. You might have to uninstall the existing version, and also `killall lutris` in case it's running in the background, but the whole process only takes a few seconds and it's fixed all the 0.5.9 bugs I've been hitting.

This is what I did in the end, but I thought I'll post it though so someone could take care of it :)

**Comment by strycore** [comment]
I have shut down the PPA because there is no one to maintain it.
Please grab the latest deb from the releases page on Github.

**Comment by BloodyIron** [comment]
> I have shut down the PPA because there is no one to maintain it. Please grab the latest deb from the releases page on Github.

So why didn't you also update the downloads section? : https://lutris.net/downloads

Having no repo/ppa for this means Lutris will never get updated. Why couldn't someone be found to maintain it, or I dunno... automate the maintenance of it?

**Comment by jar349** [comment]
@strycore Have you considered using GitHub pages to host an apt repository? Here's an idea of how that would work: https://assafmo.github.io/2019/05/02/ppa-repo-hosted-on-github.html

It should also be possible to create an Action that adds new releases to the apt repo automatically.

**Comment by philclifford** [comment]
> > Having no repo/ppa for this means Lutris will never get updated. Why couldn't someone be found to maintain it, or I dunno... automate the maintenance of it?
> 
> I volunteer to automate the maintenance of a PPA. I have never done this before, and I don't even have the faintest idea where to start, so that means that I don't know if it cannot be done, which means that it can be done.
> 
No need for a PPA if you install `deb-get`.  Once the PR there to replace the PPA with github is merged it will track and update the latest github release.

**Comment by Emanem** [comment]
@strycore appreciate all you're doing for the Linux community - please do update the download instructions for ubuntu then.

**Comment by bright-side-99** [comment]
> No need for a PPA if you install `deb-get`. Once the PR there to replace the PPA with github is merged it will track and update the latest github release.

is deb-get integrating in apt in some way? I think the benefit of having a ppa is that once set up, lutris gets updated by the OS' default update process. It seems using deb-get one would need to update lutris (and other deb-get software) separately, which does sound like avoidable overhead and therefore not ideal to me.

If my assumption is right and there would be a ppa available, I'd choose that any day.

**Comment by mbenlioglu** [comment]
> I have shut down the PPA because there is no one to maintain it.
> Please grab the latest deb from the releases page on Github.

since the .deb file is already being built and tested, it shouldn't be too complicated to make a GitHub actions trigger for when a new release is published and fully automate the uploading process to the PPA

**Comment by ehfd** [comment]
`deb-get` is giving up automatic upgrade capabilities because it doesn't use `apt-get`. To update, people would have to run `deb-get` additionally to `apt-get`. I would suggest that the [OpenSUSE](https://software.opensuse.org/download.html?project=home%3Astrycore&package=lutris) build get restored for Ubuntu at least, if the PPA is not getting restored. Either using GitHub Actions CI/CD to upload to PPAs automatically or any other `apt-get` native option should be supported.

**Comment by coaxial** [comment]
For those like me who struggle to find where the PPA is (no hits in `/etc/apt`), what worked for me was to `apt remove lutris` from my system which also remove the lutris-team/lutris PPA from the sources when running `apt update`.

**Comment by philclifford** [comment]
> is deb-get integrating with `apt update/upgrade` in some way

It facilitates using project-provided apt repositories, ppa , web sites and github repositories to get the same benefits of managed and automatable updates. The apt and ppa actions are indeed integrated so a typical usage is to `deb-get update; deb-get upgrade` which will deal with keeping all the supported deb sources up-to-date.

**Comment by philclifford** [comment]
> To upgrade, people would have to run `deb-get` additionally to `apt-get`.

Not correct: `deb-get` relies on `apt-get` for the maintenance of apt repo and ppa sources, so typically you just use `deb-get` to update/upgrade all the supported  .deb packages and the system and ppa sources will be included at the same time.

**Comment by strycore** [comment]
We're not asking all our Ubuntu users to rely on some third party tool to keep Lutris up to date. That is entirely out of the question. 
The tools to auto-publish to PPAs look interesting, I will  experiment with them.

**Comment by danieljohnson2** [comment]
If I follow you, @amedee, you don't know how to maintain a PPA, so I think the answer is 'no'.

I think you'd have to learn about it before you could help resurrect the dearly departed Lutris PPA.

**Comment by strycore** [comment]
I've maintained a PPA for over 10 years and I'm still 100% clueless on how to push packages for multiple Ubuntu series. 
So yes, any help is greatly appreciated. 
I've had a look at the tools mentioned above, `yuezk/publish-ppa-package` looks like what we're looking for but I have no idea how to set this up. 
Without some kind of assistance, this is likely to remain stuck in the current state.

**Comment by Amaylia** [comment]
@strycore
Looking at the marketplace here: https://github.com/marketplace/actions/publish-ppa-package

It looks like you need to create a .github/workflow/publish-to-lutris-team-ppa.yml file on master(?) that looks kinda like this:

```yaml
# Added this from https://github.com/marketplace/actions/publish-ppa-package
name: Publish PPA
uses: yuezk/publish-ppa-package@main
with:
    # PPA Repository to publish packages too.
    repository: 'ppa:lutris-team/lutris'
    # These need to be set up as secrets within Github.
    gpg_private_key: ${{ secrets.PPA_GPG_PRIVATE_KEY }}
    gpg_passphrase: ${{ secrets.PPA_GPG_PASSPHRASE }}
    # Points to /debian/source/format in the root of the repo.
    pkgdir: '/'

```

You'd also need the GPG private key and passphrase used for signing your PPA packages added to Github as secrets following the name pattern above: PPA_GPG_PRIVATE_KEY & PPA_GPG_PASSPHRASE.

If you look at the Bash script in the `yuezk/publish-ppa-package` repository, it grabs all of the build series' here: https://github.com/yuezk/publish-ppa-package/blob/main/build.sh#L27

And then just loops over them here https://github.com/yuezk/publish-ppa-package/blob/main/build.sh#L36-L75 to build the packages for each Ubuntu series.

I'm not really on Github too frequently, so I'm not sure how their workflows operate, but the guide here https://docs.github.com/en/actions/quickstart#viewing-your-workflow-results seems to say that once the file exists in a branch, every commit will cause that workflow to run.

I dunno if this helps, but hopefully it does!

**Comment by Amaylia** [comment]
You know, it might be easier just to integrate the same process used in https://github.com/yuezk/publish-ppa-package/blob/main/build.sh into your guys' makefile and then add it to your Travis CI process instead.

**Comment by daveyarwood** [comment]
This seems to be resolved now :tada:   I was able to do an `apt update` and `apt upgrade` and it installed Lutris 0.5.12. (I'm on Ubuntu 20.04.)

**Comment by danieljohnson2** [comment]
Sort of. @strycore reverted a change to use WebKit 4.1 after the release, and updated the .deb file - and only the .deb file- on the releases page.

I just checked, and the PPA has the original Lutris 0.5.12, not the @strycore approved version.

Hopefully anything that uses PPAs can provide an updated WebKit; it installed for me when I tried it on an Ubuntu VM.

**Comment by strycore** [comment]
The Webkit 4.1 change was for Debian and shouldn't affect recent Ubuntu versions. 
I have to see what's that about Ubuntu 18.04 no longer being supported because I certainly haven't said we were dropping support for this distro.

What is currently missing is automation. The PPA should get updated without manual intervention when a new version is released. Old versions of Lutris should never be distributed.

**Comment by Amaylia** [comment]
I just wanted to let everyone know that I was looking into the yuezk/publish-ppa-package github action, and the problem I discovered with it is that it uses a different build process than what is in Lutris's Makefile and doesn't build a functional .deb since the meson elements don't get applied.

I'm kinda working with a test PPA and a github workflow, and borrowing some things from yuezk's build scripts to make a workflow that builds using the Makefile and on multiple Ubuntu versions and then can dput each of those versions up to the PPA, but I still have some more testing to do before it will be in a state that I can make a PR for.

I'm hoping to have something a PR probably by the end of the weekend, but please let me know if I am stepping on anyone's toes here. 😬

**Comment by danieljohnson2** [comment]
@Amaylia, I feel sure that such a PR would be very welcome.

**Comment by Amaylia** [comment]
Okay, I've got some initial testing done, and managed to successfully have the GitHub workflow automatically push builds for Bionic 20.04 and Jammy 22.04 to a test PPA that I set up here: https://launchpad.net/~amaylia/+archive/ubuntu/lutris, I was able to install that package from the PPA and run it and show the dummy version bump in Lutris.. however, I'm kind of at a loss on which events in GitHub should trigger the workflow.. I was thinking probably whenever a new release is added.. but there is a opportunity here to also add testing to all commits / PRs that adds that little ✅ or ❌ next to commit messages and was just curious if, @strycore or @danieljohnson2, either of you had any interest in that.  A basic github workflow can run directive in the Makefile to set up some basic validation of commits if you like?

**Comment by Amaylia** [comment]
Here are a list of the possible events that can trigger workflows:
https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows

And this is the one I am thinking of using for now: 
https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#release

**Comment by Amaylia** [comment]
Okay, so I tested the on published release event and have that working.

I bumped the version to v0.5.12-3.
![image](https://user-images.githubusercontent.com/22217699/206949179-c99f1dda-3edb-4a32-99ec-e002d3aa7f58.png)

Created and published a release.
![image](https://user-images.githubusercontent.com/22217699/206949253-86780bc1-4d8a-4981-a341-daf43eaa6084.png)

Checked that the deb build and PPA publish ran successfully.
![image](https://user-images.githubusercontent.com/22217699/206949382-959f5bf1-01ad-42cd-8eaa-123673faa678.png)

Verified the files on my testing PPA.
![image](https://user-images.githubusercontent.com/22217699/206949791-f252f952-0fe2-425e-870b-18b23ce7f1e8.png)

I'm still waiting for them to publish, but I think I'm ready to start putting together a PR for this now.. I was thinking the other thing with testing commits and PRs could probably wait if that's something that's even desired at all.

**Comment by BloodyIron** [comment]
If I read this correctly, the PPA automation is still being built. Am I reading this correctly? If so, that's cool, and thanks for making the time for this folks! This really helps :)

But... if it is operational... can I have the details please? :3 If it's not "prod-ready" I can wait!

**Comment by Amaylia** [comment]
@BloodyIron I _think_ it's finished.. Though, @strycore has the final say on that, of course.  There is still one PR (#4653) that needs to be merged to generate the builds for Kinetic 22.10 and Lunar 23.04.

However, as of right now, you can add https://launchpad.net/~lutris-team/+archive/ubuntu/lutris or https://launchpad.net/~lutris-team/+archive/ubuntu/lutris-staging if you want the prerelease builds.  I _think_ those builds are current... but I'm not 100% sure.

Once a new release is created, we should start seeing the automated builds pushed, I think.

**Comment by BloodyIron** [comment]
Aha! I was asking since the downloads page doesn't mention the repo (for Ubuntu/etc) :) So I wasn't sure if it needed one last coat of paint before being used ;)

Thanks! I'll add the not-staging repo :D

**Comment by BloodyIron** [comment]
Hey so added the repo, and I can't upgrade...:

The following packages have unmet dependencies:
 lutris : Depends: gir1.2-webkit2-4.1 but it is not installable
E: Unable to correct problems, you have held broken packages.

Heads-up!

**Comment by Amaylia** [comment]
Sorry, @BloodyIron !  I just looked at the dates on the package on the PPA and the one in the repo, and they looked the same, but after downloading them and checking them, they do have a slight size and dependency difference.

> $ dpkg-deb -I `lutris_0.5.12~ubuntu23.04.1_all.deb`
>  new Debian package, version 2.0.
>  size 1037242 bytes: control archive=9287 bytes.
>      889 bytes,    16 lines      control              
>    25879 bytes,   294 lines      md5sums              
>      247 bytes,    12 lines   *  postinst             #!/bin/sh
>      360 bytes,    12 lines   *  prerm                #!/bin/sh
>  Package: lutris
>  Version: `0.5.12~ubuntu23.04.1`
>  Architecture: all
>  Maintainer: Mathieu Comandon <strider@strycore.com>
>  Installed-Size: 3055
>  Depends: python3:any, python3-yaml, python3-lxml, python3-requests, python3-pil, python3-gi, python3-setproctitle, python3-magic, python3-distro, python3-dbus, gir1.2-gtk-3.0, gir1.2-webkit2-4.1, psmisc, cabextract, unzip, p7zip, curl, fluid-soundfont-gs, x11-xserver-utils, mesa-utils
>  Recommends: python3-evdev, gvfs-backends, libwine-development, winetricks
>  Suggests: gamemode, gamescope
>  Section: games
>  Priority: optional
>  Homepage: https://lutris.net
>  Description: video game preservation platform
>   Lutris helps you install and play video games from all eras and from most
>   gaming systems. By leveraging and combining existing emulators, engine
>   re-implementations and compatibility layers, it gives you a central interface
>   to launch all your games.
> $ dpkg-deb -I lutris_0.5.12_all.deb 
>  new Debian package, version 2.0.
>  size 1038926 bytes: control archive=9313 bytes.
>      904 bytes,    16 lines      control              
>    25879 bytes,   294 lines      md5sums              
>      247 bytes,    12 lines   *  postinst             #!/bin/sh
>      391 bytes,    12 lines   *  prerm                #!/bin/sh
>  Package: lutris
>  Version: 0.5.12
>  Architecture: all
>  Maintainer: Mathieu Comandon <strider@strycore.com>
>  Installed-Size: 3054
>  Depends: python3:any, python3-yaml, python3-lxml, python3-requests, python3-pil, python3-gi, python3-setproctitle, python3-magic, python3-distro, python3-dbus, gir1.2-gtk-3.0, gir1.2-webkit2-4.0, gir1.2-notify-0.7, psmisc, cabextract, unzip, p7zip, curl, fluid-soundfont-gs, x11-xserver-utils, mesa-utils
>  Recommends: python3-evdev, gvfs-backends, libwine-development | libwine, winetricks
>  Suggests: gamemode, gamescope
>  Section: games
>  Priority: optional
>  Homepage: https://lutris.net
>  Description: video game preservation platform
>   Lutris helps you install and play video games from all eras and from most
>   gaming systems. By leveraging and combining existing emulators, engine
>   re-implementations and compatibility layers, it gives you a central interface
>   to launch all your games.
> 

I guess the PPA still has the version that @danieljohnson2 mentioned here: https://github.com/lutris/lutris/issues/4296#issuecomment-1340083046

Sorry to mislead you.. what version of Ubuntu are you using? gir1.2-webkit2-4.1 does seem to be available in Jammy, which is what I'm currently using.

> $ apt show -a gir1.2-webkit2-4.1
> Package: gir1.2-webkit2-4.1
> Version: 2.38.2-0ubuntu0.22.04.2
> Priority: optional
> Section: universe/introspection
> Source: webkit2gtk
> Origin: Ubuntu
> Maintainer: Ubuntu Developers <ubuntu-devel-discuss@lists.ubuntu.com>
> Original-Maintainer: Debian WebKit Maintainers <pkg-webkit-maintainers@lists.alioth.debian.org>
> Bugs: https://bugs.launchpad.net/ubuntu/+filebug
> Installed-Size: 542 kB
> Depends: gir1.2-javascriptcoregtk-4.1 (= 2.38.2-0ubuntu0.22.04.2), libwebkit2gtk-4.1-0 (= 2.38.2-0ubuntu0.22.04.2), gir1.2-gtk-3.0 (>= 3.22.0), gir1.2-soup-3.0
> Homepage: https://webkitgtk.org/
> Download-Size: 95.9 kB
> APT-Manual-Installed: no
> APT-Sources: http://security.ubuntu.com/ubuntu jammy-security/universe amd64 Packages
> Description: Web content engine library for GTK - GObject introspection data
>

**Comment by BloodyIron** [comment]
Ubuntu 20.04 LTS

**Comment by Amaylia** [comment]
Hey @BloodyIron, so I generated a new package based on https://github.com/lutris/lutris/commit/b90e3492940f2d816df22e890c6dc6bc64dad2b4 with version `0.5.12-0ubuntu1ppa1~ubuntu20.04` here: https://launchpad.net/~amaylia/+archive/ubuntu/lutris-stable/+files/lutris_0.5.12-0ubuntu1ppa1~ubuntu20.04_all.deb

This should be basically the same thing being released on GitHub right now, except the version number will allow you to keep the PPA installed since it supersedes the version of Lutris on the PPA.  Once the next release is pushed to the PPA, the PPA version will supersede the version of the package I linked and it will be upgraded.

It's not ideal, but hopefully it works for you and anyone else who is still on Focal/20.04 at the moment.

> `amaylia@neomoon-focal:~$ sudo apt update`
> Hit:1 http://us.archive.ubuntu.com/ubuntu focal InRelease
> Hit:2 http://security.ubuntu.com/ubuntu focal-security InRelease           
> Get:3 http://us.archive.ubuntu.com/ubuntu focal-updates InRelease [114 kB] 
> Hit:4 http://ppa.launchpad.net/amaylia/lutris-stable/ubuntu focal InRelease           
> Get:5 http://us.archive.ubuntu.com/ubuntu focal-backports InRelease [108 kB]          
> Hit:6 **http://ppa.launchpad.net/lutris-team/lutris/ubuntu** focal InRelease   
> Fetched 222 kB in 1s (216 kB/s)
> Reading package lists... Done
> Building dependency tree       
> Reading state information... Done
> All packages are up to date.
> 
> `amaylia@neomoon-focal:~$ sudo apt upgrade lutris`
> Reading package lists... Done
> Building dependency tree       
> Reading state information... Done
> **lutris is already the newest version (`0.5.12-0ubuntu1ppa1~ubuntu20.04`).**
> Calculating upgrade... Done
> `#`
> `# News about significant security updates, features and services will`
> `# appear here to raise awareness and perhaps tease /r/Linux ;)`
> `# Use 'pro config set apt_news=false' to hide this and future APT news.`
> `#`
> 0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
> amaylia@neomoon-focal:~$ sudo apt show -a lutris
> Package: lutris
> **`Version: 0.5.12-0ubuntu1ppa1~ubuntu20.04`**
> Priority: optional
> Section: games
> Maintainer: Mathieu Comandon <strider@strycore.com>
> Installed-Size: 3,127 kB
> Depends: cabextract, curl, fluid-soundfont-gs, gir1.2-gtk-3.0, gir1.2-notify-0.7, gir1.2-webkit2-4.0, mesa-utils, p7zip, psmisc, python3-dbus, python3-distro, python3-gi, python3-lxml, python3-magic, python3-pil, python3-requests, python3-setproctitle, python3-yaml, unzip, x11-xserver-utils, python3:any
> Recommends: gvfs-backends, libwine-development | libwine, python3-evdev, winetricks
> Suggests: gamemode, gamescope
> Download-Size: 1,037 kB
> APT-Manual-Installed: yes
> APT-Sources: http://ppa.launchpad.net/amaylia/lutris-stable/ubuntu focal/main amd64 Packages
> Description: video game preservation platform
>  Lutris helps you install and play video games from all eras and from most
>  gaming systems. By leveraging and combining existing emulators, engine
>  re-implementations and compatibility layers, it gives you a central interface
>  to launch all your games.
> 
> Package: lutris
> **`Version: 0.5.12~ubuntu20.04.1`**
> Priority: optional
> Section: games
> Maintainer: Mathieu Comandon <strider@strycore.com>
> Installed-Size: 3,128 kB
> Depends: python3:any, python3-yaml, python3-lxml, python3-requests, python3-pil, python3-gi, python3-setproctitle, python3-magic, python3-distro, python3-dbus, gir1.2-gtk-3.0, gir1.2-webkit2-4.1, psmisc, cabextract, unzip, p7zip, curl, fluid-soundfont-gs, x11-xserver-utils, mesa-utils
> Recommends: python3-evdev, gvfs-backends, libwine-development, winetricks
> Suggests: gamemode, gamescope
> Download-Size: 1,037 kB
> APT-Sources: http://ppa.launchpad.net/lutris-team/lutris/ubuntu focal/main amd64 Packages
> Description: video game preservation platform
>  Lutris helps you install and play video games from all eras and from most
>  gaming systems. By leveraging and combining existing emulators, engine
>  re-implementations and compatibility layers, it gives you a central interface
>  to launch all your games.
>

**Comment by strycore** [comment]
The build from the 0.5.12 tag will only be compatible with recent Ubuntu releases. I made a change to fix that after the release but it's not tagged as a release (no code was changed, it was a packaging issue).
If that's going to be a problem to publish packages for all Ubuntu series, we can consider making a point release with fixes to support both libwebkit 4.0 and 4.1

**Comment by danieljohnson2** [comment]
No, there were code changes involved in supporting WebKit 4.1.

Not big ones, though.

**Comment by Amaylia** [comment]
@strycore: So what's happening now is that the releases on the PPA 

> The build from the 0.5.12 tag will only be compatible with recent Ubuntu releases. I made a change to fix that after the release but it's not tagged as a release (no code was changed, it was a packaging issue). If that's going to be a problem to publish packages for all Ubuntu series, we can consider making a point release with fixes to support both libwebkit 4.0 and 4.1

@strycore: Yeah, the main problem I was trying to solve for the Ubuntu 20.04 people was just that the PPA version of the package technically supersedes the version on GitHub, due to the `~ubuntu22.04.1` bit.  So if one installs the package from the website, and has the PPA installed, the PPA will install the older build for them during an upgrade.  If you were to bump the version, that wouldn't happen anymore, and it would solve the problem for them.

@BloodyIron: It was just an option for you that would allow you to keep the PPA installed.  I don't really have a preference, I was just trying to help.  If @strycore bumps the version, then it won't be necessary at all.

**Comment by BloodyIron** [comment]
@Amaylia well I'll try the deb you built for me anyways :) Thanks!
